crypto/ocsp/ocsp_test.cc

```diff
@@ -954,6 +952,141 @@ TEST_P(OCSPRequestTest, OCSPRequestSign) {
+  const OCSPResponseSignTestVector &t = GetParam();
+
+  // Set up an empty |OCSP_BASICRESP| for signing.
+  bssl::UniquePtr<OCSP_BASICRESP> basic_response(OCSP_BASICRESP_new());
+  ASSERT_TRUE(basic_response);
+
+  // Gather certs and key needed for signing.
+  bssl::UniquePtr<X509> signer_cert(
+      CertFromPEM(GetTestData(std::string("crypto/ocsp/test/aws/" +
+                                          std::string(t.signer_cert) + ".pem")
+                                  .c_str())
+                      .c_str()));
+  ASSERT_TRUE(signer_cert);
+  bssl::UniquePtr<STACK_OF(X509)> additional_cert(CertChainFromPEM(
+      GetTestData(std::string("crypto/ocsp/test/aws/ca_cert.pem").c_str())
+          .c_str()));
+  ASSERT_TRUE(additional_cert);
+  bssl::UniquePtr<EVP_PKEY> pkey(EVP_PKEY_new());
+  if (std::string(t.private_key) == "server_key") {
+    bssl::UniquePtr<RSA> rsa(
+        RSAFromPEM(GetTestData(std::string("crypto/ocsp/test/aws/" +
+                                           std::string(t.private_key) + ".pem")
+                                   .c_str())
+                       .c_str()));
+    ASSERT_TRUE(rsa);
+    ASSERT_TRUE(EVP_PKEY_set1_RSA(pkey.get(), rsa.get()));
+  } else {
+    bssl::UniquePtr<EC_KEY> ecdsa(ECDSAFromPEM(
+        GetTestData(std::string("crypto/ocsp/test/aws/" +
+                                std::string(t.private_key) + ".pem")
+                        .c_str())
+            .c_str()));
+    ASSERT_TRUE(ecdsa);
+    ASSERT_TRUE(EVP_PKEY_set1_EC_KEY(pkey.get(), ecdsa.get()));
+  }
+
+  // Do the actual sign.
+  EXPECT_EQ(OCSP_basic_sign(basic_response.get(), signer_cert.get(), pkey.get(),
+                            t.dgst, additional_cert.get(), 0),
+            t.expected_sign_status);
+}
+
+// Test against various flags for |OCSP_basic_sign|.
+TEST(OCSPResponseSignTestExtended, OCSPResponseSign) {
+  bssl::UniquePtr<OCSP_BASICRESP> basic_response(OCSP_BASICRESP_new());
+  ASSERT_TRUE(basic_response);
+
+  bssl::UniquePtr<X509> signer_cert(CertFromPEM(
+      GetTestData(std::string("crypto/ocsp/test/aws/server_cert.pem").c_str())
+          .c_str()));
+  ASSERT_TRUE(signer_cert);
+  bssl::UniquePtr<STACK_OF(X509)> additional_cert(CertChainFromPEM(
+      GetTestData(std::string("crypto/ocsp/test/aws/ca_cert.pem").c_str())
+          .c_str()));
+  ASSERT_TRUE(additional_cert);
+
+  bssl::UniquePtr<EVP_PKEY> pkey(EVP_PKEY_new());
+  bssl::UniquePtr<RSA> rsa(RSAFromPEM(
+      GetTestData(std::string("crypto/ocsp/test/aws/server_key.pem").c_str())
+          .c_str()));
+  ASSERT_TRUE(rsa);
+  ASSERT_TRUE(EVP_PKEY_set1_RSA(pkey.get(), rsa.get()));
+
+  // Call |OCSP_basic_sign| with no flags and check the expected output.
+  EXPECT_TRUE(OCSP_basic_sign(basic_response.get(), signer_cert.get(),
+                              pkey.get(), EVP_sha256(), additional_cert.get(),
+                              0));
+  EXPECT_TRUE(
+      ASN1_TIME_check(basic_response.get()->tbsResponseData->producedAt));
```

Comment:
Can this check anything else about the time? Such as it's within the current second/minute?

Reply:
Added a time check for a 2 hour time frame